csharp-basic-topics/GetClassNameAsString/Tests/ProgramUnitTest.cs

```diff
@@ -0,0 +1,41 @@
+using Microsoft.VisualStudio.TestPlatform.TestHost;
+using System.Reflection;
+
+namespace GetClassNameAsString.Tests
+{
+    public class ProgramUnitTest
+    {
+        [Theory]
+        [InlineData(nameof(Program))]
+        public void WhenUseNameOf_ThenCorrectName(string expected)
+        {
+            var name = nameof(Program);
+            Assert.Equal(expected, name);
+        }
+
+        [Fact]
+        public void WhenUseTypeOf_ThenCorrectName()
+        {
+            string expected = "Program";
+            var name = typeof(Program).Name;
+            Assert.Equal(expected, name);
+        }
+
+        [Fact]
+        public void WhenUseGetTypeOfInstance_ThenCorrectName()
+        {
+            string expected = "Program";
+            Program p = new();
+            var name = p.GetType().Name;
+            Assert.Equal(expected, name);
+        }
+
+        [Fact]
+        public void WhenUseGetCurrentMethodDeclaringType_ThenCorrectName()
+        {
+            var expected = "ProgramUnitTest";
+            var name = MethodBase.GetCurrentMethod()!.DeclaringType!.Name;
+            Assert.Equal(expected, name);
+        }
```

Comment:
Let's introduce some newlines between Arrange/Act/Assert parts of the tests wherever applicable.